samza-core/src/main/java/org/apache/samza/zk/ZkLeaderElector.java

```diff
@@ -114,6 +100,11 @@ public boolean tryBecomeLeader() {
     String predecessor = children.get(index - 1);
     if (!predecessor.equals(currentSubscription)) {
       if (currentSubscription != null) {
+
+        // callback in case if the previous node gets deleted (when previous processor dies)
```

Comment:
What needs to be reviewed here? Did we identify a bug after this code was committed? If yes, I think it should be handled now.